packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -438,9 +440,21 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
         return this.flushDeferred.promise;
     }
 
-    protected serverDisconnectHandler = (error: IFluidErrorBase & OdspError) => {
-        this.logger.sendTelemetryEvent({ eventName: "ServerDisconnect", clientId: this.clientId }, error);
-        this.disposeSocket(error);
+    protected serverDisconnectHandler = (error: IFluidErrorBase & OdspError, clientId?: string) => {
+        // Don't dispose the socket in case a single client needs to be disconnected.
+        if (clientId === undefined) {
+            this.logger.sendTelemetryEvent({
+                eventName: "ServerDisconnect",
+                clientId: this.clientId,
+            }, error);
+            this.disposeSocket(error);
```

Comment:
It does not feel like disposeSocket() is needed here, as above code will close socket.